File: Source/Aladdin.CAPI.Environment/Environment/ConfigAuthentications.cs
using System;
using System.Xml;

namespace Aladdin.CAPI.Environment
{
    ///////////////////////////////////////////////////////////////////////////
    // Элемент описания аутентификаций
    ///////////////////////////////////////////////////////////////////////////
    [Serializable]
    public class ConfigAuthentications 
    {
        // число попыток аутентификации 
        private string attempts; 
    
        // конструктор
        public ConfigAuthentications(XmlElement element)
        {
            // получить число попыток аутентификации
            attempts = element.GetAttribute("attempts"); 
        }
        // число попыток аутентификации 
        public int Attempts { get 
        { 
            // число попыток аутентификации 
            return (attempts.Length > 0) ? Int32.Parse(attempts) : 5; 
        }}
    }
}

File: Source/Aladdin.CAPI.Environment.Java/src/aladdin/capi/environment/ConfigAuthentications.java
package aladdin.capi.environment;
import java.io.*;

///////////////////////////////////////////////////////////////////////////
// Элемент описания фабрики классов
///////////////////////////////////////////////////////////////////////////
public class ConfigAuthentications implements Serializable
{
    private static final long serialVersionUID = 5925701835596592786L;
    
    // число попыток аутентификации 
    private final int attempts; 
    
    // конструктор
    public ConfigAuthentications(int attempts) { this.attempts = attempts; }
    
    // число попыток аутентификации 
    public int attempts() { return attempts; }
}
